src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
Not sure the probability of choosing a poor hyperplane by sampling only 2 points, but if you sample 10 then it's pretty easy to find a hyperplane that splits the 10 evenly. You sample a random hyperplane, project all 10 points to the randomly chosen hyperplane, then choose a hyperplane perpendicular to this one that passes through the original splitting the 10 projected points evenly. Make sense? Easier to draw on paper then to describe here.

I'd be careful using a fully randomized Las Vegas algorithm (reject if < 80/20 split) since it costs O(n) to verify (where n is the number of points left to split in the subtree). If you end up in a loop of rejects then this can be quite costly in the early splits, but sampling more points on the initial hyperplane choice is much less costly, and if they are chosen uniformly then more or less likely to represent the full distribution.